### PR TITLE
Feature/dates refactor - REPORTS 

### PR DESCRIPTION
**Pull request by clari182**
- part of #2092 

Changes `date_downloaded`, `date_published`, `date_modified` and `date_submitted` types from string to date

Test URL: https://deploy-preview-32--clarayoudale-staging.netlify.app/

### CODE REVIEW

File: site/gatsby-site/src/utils/date.js [hunk 1]
```diff
@@ -22,3 +24,11 @@ export const isPastDate = {
...
   },
 };
+
+export const formatDate = (dateString) => {
+  const date = new Date(dateString);
+
+  return `${date.getFullYear()}-${String(date.getMonth() + 1).padStart(2, '0')}-${String(
+    date.getDate()
+  ).padStart(2, '0')}`;
+};
```
Comment (pdcp1) [inline site/gatsby-site/src/utils/date.js:34]: Do we need this custom Date format function? Maybe we can use the same that we currently using:
`import { format } from 'date-fns';`

File: site/gatsby-site/src/components/forms/IncidentReportForm.js [hunk 2]
```diff
@@ -273,22 +291,24 @@ const IncidentReportForm = () => {
...
         />
+
         <TextInputGroup
           name="date_published"
-          label="Date Published"
-          icon={faCalendar}
+          label={t('Date Published')}
           type="date"
-          placeholder="YYYY-MM-DD"
+          placeholder={t('YYYY-MM-DD')}
```
Comment (pdcp1) [inline site/gatsby-site/src/components/forms/IncidentReportForm.js:299]: Do we have translations for `YYYY-MM-DD` ? If not, please add

**Review by pdcp1 — CHANGES_REQUESTED** [review]

**Review by pdcp1 — CHANGES_REQUESTED** [review]
I tried to add a variant on the preview environment https://deploy-preview-32--clarayoudale-staging.netlify.app/ and it failed because of an error on the date field:

```
{
    "data": {
        "createVariant": null
    },
    "errors": [
        {
            "message": "reason=\"role \\\"role is admin\\\" in \\\"aiidprod.reports\\\" does not have insert permission: could not validate document: \\n\\tdate_downloaded: Invalid type. Expected: type: undefined, bsonType: date, given: [string mixed]\\n\\tdate_modified: Invalid type. Expected: type: undefined, bsonType: date, given: [string mixed]\\n\\tdate_published: Invalid type. Expected: type: undefined, bsonType: date, given: [string mixed]\\n\\tdate_submitted: Invalid type. Expected: type: undefined, bsonType: date, given: [string mixed]\"; code=\"SchemaValidationFailedWrite\"; untrusted=\"insert not permitted\"; details=map[]",
            "locations": [
                {
                    "line": 2,
                    "column": 3
                }
            ],
            "path": [
                "createVariant"
            ]
        }
    ]
}
```

image

**Review by pdcp1 — CHANGES_REQUESTED** [review]
We should change the formated date for Variants (ie: https://deploy-preview-32--clarayoudale-staging.netlify.app/cite/568/)

image

**Review by pdcp1 — CHANGES_REQUESTED** [review]
On https://deploy-preview-32--clarayoudale-staging.netlify.app/apps/variants/ when I tried to edit (also if trying to `Approve` or `Reject`) a Variant I got a similar error with the dates:

```
{
    "data": null,
    "errors": [
        {
            "message": "Variable \"$set\" got invalid value {\"date_modified\":\"2023-11-20\",\"date_published\":\"2023-11-13\",\"epoch_date_modified\":1700515791,\"inputs_outputs\":[\"dsdsd\",\"saaaaaaaaaaaaaa\",\"dadasdasdadasd\"],\"submitters\":[\"saildiaokmj\"],\"text\":\"u jhju gu ju jju guu jhju gu ju jju guu jhju gu ju jju guu jhju gu ju jju guu jhju gu ju jju guu jhju gu ju jju guu jhju gu ju jju gu\"}.\nIn field \"date_modified\": Expected type \"DateTime\", found \"2023-11-20\".\nIn field \"date_published\": Expected type \"DateTime\", found \"2023-11-13\".",
            "locations": [
                {
                    "line": 1,
                    "column": 51
                }
            ]
        }
    ]
}
```

image

**Review by pdcp1 — CHANGES_REQUESTED** [review]
When I edit and save the *Incident Date* field on a Variant, it doesn't display the correct value. It displays always `dd/mm/yyyy`.

https://deploy-preview-32--clarayoudale-staging.netlify.app/apps/variants/

image

**Review by pdcp1 — APPROVED** [review]

**Comment by clari182** [comment]
@cesarvarela can we merge this please?

**Review by cesarvarela — CHANGES_REQUESTED** [review]
Something going on with how dates are displayed on citation pages, prod vs preview:

image

image

**Comment by cesarvarela** [comment]
@clari182 remember to whitelist the preview so we can test on it.

**Comment by kepae** [comment]
(I'm also testing this before merging, given the refactor scale.)

File: site/gatsby-site/src/components/variants/VariantList.js [hunk 3]
```diff
@@ -68,7 +69,9 @@ const VariantCard = ({ variant, incidentId }) => {
             <div className="font-bold">
               <Trans>Incident Date</Trans>:
             </div>
-            <div>{variant.date_published}</div>
+            <div>
```
Comment (kepae) [inline site/gatsby-site/src/components/variants/VariantList.js:72]: <div> should also be inside conditional?

File: site/gatsby-site/src/pages/apps/reports.js [hunk 4]
```diff
@@ -32,6 +32,9 @@ const query = gql`
         epoch_date_downloaded
```
Comment (kepae) [inline site/gatsby-site/src/pages/apps/reports.js:32]: these `epoch_date_` fields should no longer be requested or needed, right? or do you plan to remove them in a followup PR, along with updating the schema.json?

**Review by kepae — APPROVED** [review]
LGTM, separate from from my questions. @clari182 please ping me when you want to merge this.

File: site/gatsby-site/src/pages/apps/reports.js (hunk 4 above)
Comment (clari182) [reply site/gatsby-site/src/pages/apps/reports.js:32]: I plan to remove them in a follow up otherwise it was too many dangerous changes in one PR.

**Review by pdcp1 — APPROVED** [review]

**Comment by clari182** [comment]
@cesarvarela preview is whitelisted, realm up to date and the DB updated with dates, if you can, can you review this PR?
Thanks!!

**Review by cesarvarela — APPROVED** [review]

**Comment by cesarvarela** [comment]
I don't know if there is something else we want to merge before this, but I'm okay with ripping this band-aid.

**Comment by kepae** [comment]
Merged; it's time!

**Pull request merged**